presto-raptor/src/main/java/com/facebook/presto/raptor/metadata/DatabaseShardManager.java

```diff
@@ -537,6 +541,9 @@ public void createBuckets(long distributionId, int bucketCount)
                 if (gracePeriod && (nanosSince(startTime).compareTo(startupGracePeriod) < 0)) {
                     throw new PrestoException(SERVER_STARTING_UP, "Cannot reassign buckets while server is starting");
                 }
+                if (gracePeriod) {
+                    assignmentLimiter.checkAssignFrom(nodeId);
```

Comment:
That variable means whether or not the caller wants to use the grace period check. Currently only the tests use it to skip it. It's more needed for the node assignments. I could refactor this code to remove it.